pkg/errors/errors.go

```diff
@@ -0,0 +1,75 @@
+package errors
+
+import (
+	"fmt"
+)
+
+// AtmosError represents a structured error with rich context.
+type AtmosError struct {
+	Base     error                  // Underlying error for unwrapping
+	Message  string                 // Human-readable message
+	Meta     map[string]interface{} // Contextual metadata
+	Tips     []string               // User guidance
+	ExitCode int                    // Process exit code
```

Comment:
I don't think we have the functions to set this. The idea here is we need to be able to return an error with a suggested exit code. Then if we `log.Fatal(ae)` it will use that exit code if available.

Reply:
This can be in a separate PR, but then we should remove the used variable.